lib/rspec/core/shared_example_group.rb

```diff
@@ -213,16 +213,35 @@ def valid_name?(candidate)
 
         def warn_if_key_taken(context, key, new_block)
           existing_module = shared_example_groups[context][key]
-
           return unless existing_module
 
-          RSpec.warn_with <<-WARNING.gsub(/^ +\|/, ''), :call_site => nil
-            |WARNING: Shared example group '#{key}' has been previously defined at:
-            |  #{formatted_location existing_module.definition}
-            |...and you are now defining it at:
-            |  #{formatted_location new_block}
-            |The new definition will overwrite the original one.
-          WARNING
+          old_definition = formatted_location existing_module.definition
+          new_definition = formatted_location new_block
+
+          if old_definition == new_definition
```

Comment:
I think that `old_definition == new_definition` is a poor way to detect this.  There are other reasons why `old_definition == new_definition` could evaluate to `true` (e.g. the user is using `load` instead of `require` to load the file from two places).  It would be a lot more precise to see if the file is included in `RSpec.configuration.loaded_spec_files`.  If so, then we know that the file is named in such a way to match the pattern and be loaded by RSpec itself, and there's no need to use qualifying language like "you might be seeing this warning because of a naming issue" or "if you have named the file to end in _spec.rb`.